modules/thermo-cycler/thermo-cycler-arduino/thermo-cycler-arduino.ino

```diff
@@ -702,6 +702,27 @@ void read_from_serial() {
 /////////////////////////////////
 /////////////////////////////////
 
+void front_button_callback()
+{
+  if(digitalRead(PIN_FRONT_BUTTON_SW) == LOW)
+  {
+    front_button_pressed = true;
+  }
+}
+
+void front_button_check()
+{
+  if (front_button_pressed)
+  {
+    delay(100); // debounce
```

Comment:
This is a really long delay for something that will block the mainloop. 

We could probably fix this by changing the logic to
- Save the last time the button was hit
- When this is called, check to see if it's been less than some timeout since then, and don't do anything if so (to prevent this from getting called multiple times by the bounce)
- If it's been long enough that we should actuate again, write the current millis() to a variable somewhere and actuate the lid

Reply:
Totally. I think I was in a hurry to just add some functionality to test the button.
I'll make these changes